src/main/java/com/adorsys/keycloakstatuslist/service/StatusListService.java

```
@@ -0,0 +1,142 @@
+package com.adorsys.keycloakstatuslist.service;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.classic.methods.HttpPut;
+import org.apache.hc.client5.http.config.RequestConfig;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.util.Timeout;
+import org.jboss.logging.Logger;
+
+import com.adorsys.keycloakstatuslist.config.StatusListConfig;
+import com.adorsys.keycloakstatuslist.model.TokenStatus;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+public class StatusListService {
+    private static final Logger logger = Logger.getLogger(StatusListService.class);
+    private static final ObjectMapper objectMapper = new ObjectMapper()
+            .registerModule(new JavaTimeModule());
+
+    private final StatusListConfig config;
+
+    public StatusListService(StatusListConfig config) {
+        this.config = config;
+    }
+
+    /**
+     * Publishes a new token status to the status list server (POST).
+     */
+    public boolean publishTokenStatus(TokenStatus tokenStatus) {
+        if (!config.isEnabled()) {
+            logger.debug("Status list service is disabled. Skipping token status publication.");
+            return false;
```

You may consider throwing exceptions instead of returning booleans. That's what they are made for.